benches/wasm-kernel/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+#![no_std]
+#![feature(lang_items)]
+#![feature(core_intrinsics)]
+
+extern crate rlibc;
+extern crate tiny_keccak;
+
+use tiny_keccak::Keccak;
+
+#[no_mangle]
+#[lang = "panic_fmt"]
+pub extern "C" fn panic_fmt(
+	_args: ::core::fmt::Arguments,
+	_file: &'static str,
+	_line: u32,
+	_col: u32,
+) -> ! {
+	use core::intrinsics;
+	unsafe {
+		intrinsics::abort();
+	}
+}
+
+pub struct TinyKeccakTestData {
+	data: &'static [u8],
+	result: &'static mut [u8],
+}
+
+#[no_mangle]
+pub extern "C" fn prepare_tiny_keccak() -> *const TinyKeccakTestData {
+	static DATA: [u8; 4096] = [254u8; 4096];
+	static mut RESULT: [u8; 32] = [0u8; 32];
+
+	static mut TEST_DATA: Option<TinyKeccakTestData> = None;
+
+	unsafe {
+		if let None = TEST_DATA {
+			TEST_DATA = Some(TinyKeccakTestData {
+				data: &DATA,
+				result: &mut RESULT,
+			});
+		}
+		TEST_DATA.as_ref().unwrap() as *const TinyKeccakTestData
+	}
+}
+
+#[no_mangle]
+pub extern "C" fn bench_tiny_keccak(test_data: *const TinyKeccakTestData) {
+	unsafe {
+		let mut keccak = Keccak::new_sha3_256();
```

Comment:
Not exactly sure what it is we're benchmarking here, but just thought I'd point out that this is *not* the keccak used in parity; this one uses the `0x06` delimiter whereas the "normal" keccak 256 uses `0x01`.

Reply:
Nice to know, thanks! 

TBH I didn't bothered too much and I just have ~ported~^W stolen the bench from [tiny-keccak repo](https://github.com/debris/tiny-keccak/blob/7d8d499f8027044b38f73b30d130c36b6859faed/benches/sha3.rs#L16-L26) 😄

Reply:
better to use keccak224-512, it's not regular sha3

Reply:
Could you expand on that? I really have no idea why keccak224-512 is better over this sha3-256 for benchmarking

Reply:
Changed to `keccak256`!